Feature/about us

vatakusi opened the pull request:
2 секции: "О нас" и "Контакты"

vatakusi committed:
include-section-aboutus

vatakusi committed:
between version

vatakusi committed:
about-us+contacts

2 sections: about-us+contacts

vatakusi committed:
Merge branch 'main' into feature/about-us

Persatyi commented:
Круто сделано

OlenaPatlata commented on src/index.html:
```diff
@@ -20,6 +20,9 @@
     <!-- Так добавляются фрагменты в главные HTML-файлы страниц. -->
+
+    <include src="./partials/about-us.html"></include>
+    <include src="./partials/contacts.html"></include>
```
Секцию contacts нужно подключить после секции request, как указано на макете.

OlenaPatlata commented on src/partials/about-us.html:
```diff
@@ -0,0 +1,61 @@
+<section class="aboutus section">
+  <div class="container">
+    <div class="aboutus__title-wrap">
+      <h2 class="aboutus__title">о нас</h2>
```
Добавить для h2 общий класс  title, в котором прописаны общие стили для заголовков.

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+      height: 200px;
+    }
+
+    &-pitch {
```
Заменить стили для класса .aboutus__items-pitch (так как в карточках разный отступ от текста до низа карточки, что зависит от того одна строчка текста или две)
Убрать: padding-bottom: 39px;- когда две строчки текста его выпирает из карточки.
Добавить:
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    align-content: center;
    height: 100%;

vatakusi replied:
точно, выпирает, но достаточно убрать только padding-bottom: 39px и выпадение снизу уходит; остальное (флекс и ниже для добавления) "ломает" внутри items структуру.

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+      }
+    }
+
+    &-three {
```
Для класса .aboutus__title-three убрать из стилей:
    max-width: 250px; 
    margin: auto;
В карточках в mobile разный отступ от этого текста до низа карточки, что зависит от того одна строчка текста или две, ниже также указано про необходимость внесения изменений в стили для класса .aboutus__items-pitch.

vatakusi replied:
max-width: 250px; необходим для правильного переноса текста, иначе лишние слова остаются в строке вместо переноса как на макете.

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+
+  &__items {
+    height: 160px;
+    width: 290px;
```
Убрать в стилях    для класса .aboutus__items ширину 290рх, ширину не ограничиваем. Ширину  вычисляем используя калькулятор  начиная с таблетки width: calc(100% / 2 - 30px); начиная с десктопа     width: calc(100% / 3 - 30px). То есть убрать все ширины в пикселях, от 0 и в мобилке ширина 100%, дальше только через calc.

vatakusi replied:
ок, поменял, но не совсем понял приоритет такого подхода, поскольку calc() сам по себе нагружает, хоть и незначительно, вычислительные ресурсы. В данной правке мы приходим к той-же ширине что задана, только путем допвычисления.

OlenaPatlata replied:
Calc отлично работает вместе с отрицательными марджинами, если не используете отрицательные марджины, то и calc не нужен.

vatakusi replied:
ок, чтобы не бегать туда-сюда два раза, оставляю "калк"

OlenaPatlata replied:
Повторяю, эта формула работает только в комплексе с отрицательными марджинами у родителя, поэтому убираем calc, возвращаем width.

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+    }
+
+    &-icon {
+      display: block;
```
В общих стилях _reset.scss уже задано для всех svg свойство display: block,поэтому здесь нужно убрать display: block.

OlenaPatlata replied:
?

vatakusi replied:
я его не вижу у себя в vccode. Я его не задавал.
display: block по умолчанию идет если не указано иное.

OlenaPatlata commented on src/partials/about-us.html:
```diff
@@ -0,0 +1,61 @@
+      <li class="aboutus__items">
+        <div class="aboutus__items-pitch">
+          <svg class="aboutus__items-icon">
+            <use href="images/sprite.svg#support-1" width="50" height="50"></use>
```
Убрать высоту и ширину, они заданы в стилях.

OlenaPatlata replied:
Нужно убрать высоту и ширину для иконки иначе она не будет менять размеры на таблетке и не станет 60/60

vatakusi replied:
убрал, сейчас еще раз "запулю"

OlenaPatlata commented on src/partials/about-us.html:
```diff
@@ -0,0 +1,61 @@
+      <li class="aboutus__items">
+        <div class="aboutus__items-pitch">
+          <svg class="aboutus__items-icon">
+            <use href="images/sprite.svg#manufacturing-1" width="50" height="50"></use>
```
Убрать высоту и ширину, они заданы в стилях.

OlenaPatlata replied:
Нужно убрать высоту и ширину для иконки иначе она не будет менять размеры на таблетке и не станет 60/60

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+    padding-bottom: 130px;
+  }
+
+  &__title {
```
С 23 по 55 строку убрать все стили, они полностью дублируют стили прописанные для класса title (то что в общих стилях title  задано mardgin-top 20 рх только для мобилки я вижу, Алексей это исправит - нужно от 0 и до 767 рх).

vatakusi replied:
закомментил

OlenaPatlata replied:
Стили для title написаны, но пока не подключены в main.scss, поэтому не применяются. Алексей закончит свою ветку и тогда подключит их в main.scss.

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+  }
+
+  &__descript {
+    margin-bottom: 20px;
```
Убрать марджин, толкать вниз должен родитель div с классом .aboutus__title-wrap, который оборачивает h2 и p.

OlenaPatlata replied:
Почему не убрали margin-bottom: 20px;? Этот марджин пробивает родителя на мобильной версии, а на остальных версиях добавляется к марджину родителя, что увеличивает отступ на лишние 20рх.

vatakusi replied:
убрал, видимо старая версия. Сейчас "запулю" новую.

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+      @include font(40px, 1.5, bold);
+    }
+
+    &-wrap {
```
Добавить сразу марджин нижний 20 рх, чтобы оттолкнуть первую карточку.

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+    }
+
+    &-wrap {
+      @include mq(tablet) {
```
Изменить нижний марджин на 50рх согласно макета.

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+      margin-right: 0;
+      margin-bottom: 0;
+
+      &:nth-child(-n + 3) {
```
Лучше дать родителю ul отрицательный марджин, а  детям li положительные марджины, тогда не нужно использовать разные вариации nth-child(-n + 3). Особенность с отрицательными марджинами только в таблетке, там верх и низ по 20, а лево-право по 30рх, значит родителю ul
    margin-bottom: -10px;
    margin-top: -10px;
    margin-left: -15px;
    margin-right: -15px;
А детям li
    margin-bottom: 10px;
    margin-top: 10px;
    margin-right: 15px;
    margin-left: 15px;
Начиная с десктопа у родителя     margin: -15px; у детей     margin: 15px;

vatakusi replied:
отрицательные марджины хороши на скорую руку. Текущий подход мне кажется более точечным, оставлю его с Вашего разрешения ))

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+
+    &-icon {
+      display: block;
+      margin: auto;
```
margin: auto; убрать, центрировать флексом

vatakusi replied:
в этой версии к классу флекс не примениться. Только через родителя aboutus__items-pitch {align-items: center;} который в мобильной версии блочный и чтобы перенести туда выравнивание нужно дополнительно задать ему флекс. Одна строка выглядит логичнее, поскольку дисплей: блок; по умолчанию

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+    &-icon {
+      display: block;
+      margin: auto;
+      margin-bottom: 15px;
```
margin-bottom: 15px; по макету только до таблетки, добавить условия что в таблетке уже 30рх.

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+      padding-right: 20px;
+      padding-left: 20px;
+
+      @include mq(tablet) {
```
Добавить обязательно к флексу:
    justify-content: center;
    width: 100%;

vatakusi replied:
добавил, но чудесно работает без него

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,189 @@
+
+    &-pitch {
+      padding-top: 39px;
+      padding-bottom: 39px;
```
Убрать

OlenaPatlata commented on src/sass/layouts/_contacts.scss:
```diff
@@ -0,0 +1,189 @@
+  }
+
+  @include mq(tablet) {
+    padding-top: 90px;
```
Добавить padding-bottom: 90px;

OlenaPatlata replied:
Не добавили padding-bottom: 90px для класса .contacts и на таблетке нижний паддинг остался с мобилки 40рх,чтоне соответствует макету, нужно 90рх.

vatakusi replied:
видимо все таки старая версия

vatakusi replied:
добавил

OlenaPatlata commented on src/partials/contacts.html:
```diff
@@ -0,0 +1,73 @@
+<section class="contacts">
+  <div class="contacts container">
```
Из-за того что у секции и контейнера задан одинаковый класс contacts произошло дублирование стилей секции на контейнер и сверху теперь два паддинга по 90рх и у секции и у контейнера (чтобы меньше править кода убрать вообще класс contacts у секции, а добавить общий класс section).

vatakusi replied:
переназвал класс: "contacts-" и добавил общий класс "section"

OlenaPatlata replied:
Нельзя назвать класс .contacts-, пусть будет .contacts-section и общий класс .section

vatakusi replied:
длинные классы сбивают меня с толку, ну ок, переназвал ))

OlenaPatlata replied:
Это Вы еще не видели как названы классы у Вадима и Алексея на слайдерах.

vatakusi replied:
вернул старое название поскольку фон получается уже не "резиновый" и переопределять по БЭМу не понятно зачем. Конейнер имеет глобальную настройку с цветом фона для всех форматов

OlenaPatlata commented on src/sass/layouts/_contacts.scss:
```diff
@@ -0,0 +1,189 @@
+    }
+
+    & li {
+      margin-bottom: 10px;
```
Добавить условие в 79 строке, что margin-bottom: 10px кроме последнего li

OlenaPatlata commented on src/sass/layouts/_contacts.scss:
```diff
@@ -0,0 +1,189 @@
+  background-color: var(--background-dark);
+
+  &__text-wrap {
+    display: block;
```
Добавить margin-bottom: 15px; чтобы оттолкнуть карту.

vatakusi replied:
в мобильной версии этот зазор задан в селекторе всего списка адресов - contacts__list

OlenaPatlata replied:
Задав классу .contacts__list margin-bottom: 15px вы пробили двух родителей address class="contacts__adress" и div class="contacts__text-wrap". 
Отталкиваться должны соседние элементы, значит div class="contacts__text-wrap" должен оттолкнуть нижним марджином div class="contacts__map".

vatakusi replied:
ок, исправил

OlenaPatlata commented on src/sass/layouts/_contacts.scss:
```diff
@@ -0,0 +1,189 @@
+    @include mq(tablet) {
+      display: flex;
+      /* justify-content: space-around; */
+      margin-bottom: 60px;
```
Исправить в таблетке на margin-bottom: 50px согласно макету, с десктопа уже дать 60рх, для hd - 70рх.

vatakusi replied:
поправил))

OlenaPatlata commented on src/sass/layouts/_contacts.scss:
```diff
@@ -0,0 +1,189 @@
+    }
+
+    @include mq(desktop) {
+      width: 100%;
```
Добавить margin-bottom: 60px; а для hd уже margin-bottom:70рх.

OlenaPatlata replied:
Забыли добавить margin-bottom: 60px; а для hd уже margin-bottom:70рх.

- У меня съехали строки, пока не могу понять о каком это классе идет речь

vatakusi replied:
поправил

OlenaPatlata commented on src/partials/contacts.html:
```diff
@@ -0,0 +1,73 @@
+      </div>
+    </article>
+    <article class="rekviz">
+      <div class="rekviz__title-wrap">
```
Зачем добавляли div с классом .rekviz__title-wrap, если потом он в стилях вообще не участвует? 
Можно использовать этот div как родитель с позицией relative для иконки (кружок и стрелка вверх) с позицией абсолют.

vatakusi replied:
иначе border расползается в  мобильное версии - по горизонтали, в остальных по вертикали. А так wrap задает ему форму независимо от экрана и влияния display: block; 
Добавил position: relative для иконки-свайпера

OlenaPatlata commented on src/sass/layouts/_contacts.scss:
```diff
@@ -0,0 +1,189 @@
+    }
+  }
+
+  &__title {
```
Аналогично как и для секции "О нас" убрать общие стили с класса .contacts__title, они повторяют класс .title. Дать классу .contacts__title только стили внешней геометрии.

vatakusi replied:
ок, только почему то общий title не срабатывает

OlenaPatlata replied:
Стили для title написаны, но пока не подключены в main.scss, поэтому не применяются. Алексей закончит свою ветку и тогда подключит их в main.scss.

vatakusi committed:
fix-review-comments

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,209 @@
+      display: flex;
+      justify-content: center;
+      /* max-width: 324px; */
+      width: calc(100% / 2 - 30px);
```
Нужно исправить, если не используем отрицательные марджины, тогда  убираете calc и ставите width: 324px.

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,209 @@
+
+      @include mq(tablet) {
+        @include font(16px, 1.18, 600);
+      }
```
Для hd размер шрифта уже 20рх, нужно добавить ещё одно правило.

vatakusi replied:
поправил, у меня макете правда - 22px;

OlenaPatlata commented on src/sass/layouts/_about-us.scss:
```diff
@@ -0,0 +1,209 @@
+    @include mq(desktop) {
+      position: relative;
+      /* width: 290px; */
+      width: calc(100% / 3 - 30px);
```
Без отрицательных марджинов смысла в использовании calc  нет, так что возвращайте width: 290px; и убирайте  width: calc(100% / 3 - 30px);.

OlenaPatlata replied:
Повторяю: возвращайте width: 290px; и убирайте width: calc(100% / 3 - 30px), без отрицательных марджинов эта формула работает не правильно и тогда ширина карточек меньше чем на макете.

vatakusi committed:
fix-review-comments

OlenaPatlata reviewed (changes requested):
Нужно чтобы Вы удалили лишние спрайты, так в html у Вас подключены иконки из sprite.svg, но также Вы добавили еще два спрайта symbol-defs.svg (один в папке images, другой в папке about-us-icon. Если в этих спрайтах есть иконки, которые будете использовать, то скопируйте и перенесите их в sprite.svg.

vatakusi committed:
fix-2-after-review

OlenaPatlata reviewed (changes requested):
Пару правок и будет готово.

vatakusi committed:
fix_little

вернул класс "contacts", вместо "contacts-section", поскольку теряется цвет фона в "резиновой версии".

vatakusi committed:
fix_fin

OlenaPatlata committed:
Merge branch 'main' into feature/about-us
